pkg/retrieval/retrieval.go

```diff
@@ -183,20 +185,23 @@ func (s *Service) closestPeer(addr swarm.Address, skipPeers []swarm.Address) (sw
 
 func (s *Service) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream) error {
 	w, r := protobuf.NewWriterAndReader(stream)
-	defer stream.Close()
+	defer stream.FullClose()
 	var req pb.Request
 	if err := r.ReadMsg(&req); err != nil {
+		_ = stream.Reset()
```

Comment:
this can be also handled inside a defer statement in case you have a named return

Reply:
Done